### ISSUE TITLE
Requirements for other FLOSS projects?

### ISSUE THREAD

**Issue by luzpaz** [alexlarsson/nightly-build-apps#1]
Are there any requirements for including other LibreGraphics projects?
- Is it only for gtk based programs? ie Qt5  

CC @aoloe

**Comment by alexlarsson** [comment]
The runtime we use has gtk2 and gtk3, but no qt. However you could easily Build qt as a dependency like the others. Should not be too hard, as at is pretty standalone.

I don't plan to work on other builds ATM, but if anyone else does so I will gladly add them to the build.

**Comment by aoloe** [comment]
very interesting project!

three questions:
- is there an how-to for creating the .def files? (is https://wiki.gnome.org/Projects/SandboxedApps _the_ how-to?)
- any plan to get debian to have a package for it?
- does it only work when gnome is installed? (the ubuntu package seems to say no...)

**Comment by alexlarsson** [comment]
The .def file are just shell scripts that are sourced by the build.sh script. The entire thing is nothing official, just a shell script i hacked up in a day.

I haven't talked to debian about packaging xdg-app. Do you know a good person to talk to?

There is really no dependency on gnome in xdg-app per se, all it requires is a recent kernel, and fuse support. Each individual runtime (like the gnome one) can have requirements on the desktop running on the host though. For instance, gnome apps that run non-sandboxed will probably need a dconf running to get persistent configuration stored.

**Comment by DoomHammer** [comment]
@aoloe A bit of shameless plug but you can try [doomhammer/homebrew-xdg](https://github.com/doomhammer/homebrew-xdg) to get xdg-app on Debian (that's what I do on Linux Mint). Needs a bit of manual work, but other than that it works and installs to your home directory only.

**Comment by alexlarsson** [comment]
Well, i have xdg-app debian packages on the wiki, but I guess this was about official packages.

**Comment by DoomHammer** [comment]
Yeah, that would be great but knowing how packagers in different distros are overworked it may not come too soon.

**Comment by aoloe** [comment]
i guess that the first step to acceptance for xdg-app, is to be included in the official repositories.
it's not about me getting it (i can compile it, no problem :-) but if we create a package i want our users to be able to install it with very little effort.
otherwise, we already have a ppa or people can follow some instructions and compile by themselves :-)

**Comment by alexlarsson** [comment]
I've updated the nightly build to now also have mypaint and darktable. It also has scribus, but there seems to be some issue with it, because its not working well for me.

In have also changed the way these are built so that it should be easier to add other apps too.
